mdetsims/sim_utils.py

```diff
@@ -30,8 +30,12 @@ class Sim(object):
 
     Parameters
     ----------
+    fraction : float, 0 to 1
+        The fraction of galxies with g1/g2 shear.
     rng : np.random.RandomState
         An RNG to use for drawing the objects.
+    sel : np.random.RandomState
```

Comment:
Why do we need an extra RNG here? Can't we use the one we have?

Reply:
After lifting the branching (with flag_ex condition), the issue persists. So, I continue to use an independent Rng here.